specifyweb/frontend/js_src/lib/components/TreeView/Row.tsx

```diff
@@ -141,7 +143,10 @@ export function TreeRow({
     []
   );
 
-  return (
+  const hasNoChildrenNodes =
+    nodeStats?.directCount === 0 && nodeStats.childCount === 0;
+
+  return showApplicableNodes && hasNoChildrenNodes ? null : (
```

Comment:
nodeStats are loaded async, right?

can you double check that the nodeStats are always loaded by the time this condition is evaluated for child rows? i.e does it cause nodes to show a loading indicator while node stats are fetching, and then disappear from view once server responds with `.chldCount === 0`?

if that is the case, that is a rather bad UX - it would perhaps be better to gray out those nodes instead

Reply:
@maxxxxxdlp , indeed the nodeStats can be undefined at first when loading. I thought adding a condition that if nodeStat is undefined && hideEmptyNodes is true => add the class 'opacity-50 pointer-events-none'. The problem is that the nodes for which we never display the counts will always have a nodeStat undefined. What would be your go to in this case?